front/app/containers/Admin/projects/components/ProjectRow/ProjectMoreActionsMenu.tsx

```diff
@@ -33,14 +38,27 @@ const ProjectMoreActionsMenu = ({ projectId, setError }: Props) => {
     }
   };
 
+  const setLoadingState = (
+    type: 'deleting' | 'copying',
+    isLoading: boolean
+  ) => {
+    if (type === 'copying') {
+      setIsCopying(isLoading);
+    } else if (type === 'deleting') {
+      setIsDeleting(isLoading);
```

Comment:
I wonder if we should move this state to the parent, because we use a variant of "is deleting" there too.

Reply:
That would couple them too much. This way this component handles it's own business and only provides a state when needed

Reply:
Making the changes as suggested. Thinking about it more. We probably won't use this any other use case

Reply:
While implementing it, it doesn't feel right. This component handles the actions and as such should handle the states related to those actions. If we pull that out, we are coupling them in an unhealthy way. It would mean four props for just two actions. If we want to handle another action, that is two more props and the parent would need to know about all of them. I think it is better to pass action types and one loading state. The parent component can always choose which action types to handle if there is need, leaving these two components with fewer reasons to change